diff --git a/node_modules/@loopback/rest-crud/dist/crud-rest.controller.js b/node_modules/@loopback/rest-crud/dist/crud-rest.controller.js
index 0393439..78a9090 100644
--- a/node_modules/@loopback/rest-crud/dist/crud-rest.controller.js
+++ b/node_modules/@loopback/rest-crud/dist/crud-rest.controller.js
@@ -58,6 +58,7 @@ function defineCrudRestController(modelCtor, options) {
         (0, rest_1.get)('/', {
             ...response.array(200, `Array of ${modelName} instances`, modelCtor, {
                 includeRelations: true,
+                exclude: modelCtor.getHiddenProperties(),
             }),
         }),
         tslib_1.__param(0, rest_1.param.filter(modelCtor)),
@@ -69,6 +70,7 @@ function defineCrudRestController(modelCtor, options) {
         (0, rest_1.get)('/{id}', {
             ...response.model(200, `${modelName} instance`, modelCtor, {
                 includeRelations: true,
+                exclude: modelCtor.getHiddenProperties(),
             }),
         }),
         tslib_1.__param(0, (0, rest_1.param)(idPathParam)),
