diff --git a/node_modules/@loopback/rest/dist/validation/ajv-factory.provider.js b/node_modules/@loopback/rest/dist/validation/ajv-factory.provider.js
index c52ef52..3070581 100644
--- a/node_modules/@loopback/rest/dist/validation/ajv-factory.provider.js
+++ b/node_modules/@loopback/rest/dist/validation/ajv-factory.provider.js
@@ -43,6 +43,10 @@ let AjvFactoryProvider = class AjvFactoryProvider {
             const ajvInst = new ajv_1.default(ajvOptions);
             ajvInst.addKeyword('components');
             ajvInst.addKeyword('x-typescript-type');
+            ajvInst.addKeyword('x-precision');
+            ajvInst.addKeyword('x-scale');
+            ajvInst.addKeyword('x-generated');
+            ajvInst.addKeyword('x-length');
             (0, ajv_keywords_1.default)(ajvInst, validationOptions.ajvKeywords);
             (0, ajv_errors_1.default)(ajvInst, validationOptions.ajvErrors);
             if (this.keywords) {
