diff --git a/node_modules/loopback-connector-mysql/lib/discovery.js b/node_modules/loopback-connector-mysql/lib/discovery.js
index dab2abe..1ac3951 100644
--- a/node_modules/loopback-connector-mysql/lib/discovery.js
+++ b/node_modules/loopback-connector-mysql/lib/discovery.js
@@ -353,7 +353,27 @@ function mixinDiscovery(MySQL, mysql) {
 
     return sql;
   };
+  /**
+   * Discover unique keys for a given table
+   * @param {String} table The table name
+   * @param {Object} options The options for discovery
+   */
 
+  /*!
+   * Retrieves a list of column names that have unique key index
+   * @param schema
+   * @param table
+   * @returns {string}
+   */
+  MySQL.prototype.buildQueryUniqueKeys = function(schema, table) {    const sql = 'SELECT Column_name AS "columnName",' +
+    ' table_schema AS "owner",' +
+    ' table_name AS "tableName"' +
+    ' FROM Information_schema.statistics' +
+    ' WHERE Table_schema = ' + mysql.escape(schema) +
+    ' AND Table_name = ' + mysql.escape(table) +
+    ' AND Non_unique = 0 AND Index_name <> \'PRIMARY\';';
+    return sql;
+  };
   /**
    * Discover foreign keys that reference to the primary key of this table
    * @param {String} table The table name
