diff --git a/node_modules/loopback-datasource-juggler/lib/datasource.js b/node_modules/loopback-datasource-juggler/lib/datasource.js
index c19974b..0ca0ec4 100644
--- a/node_modules/loopback-datasource-juggler/lib/datasource.js
+++ b/node_modules/loopback-datasource-juggler/lib/datasource.js
@@ -1733,6 +1733,14 @@ DataSource.prototype.discoverSchemas = function(tableName, options, cb) {
       if (uniqueKeys.includes(propName)) {
         schema.properties[propName]['index'] = {unique: true};
       }
+      // set useDefaultIdType: false in the case of id property with generated: 1 and has string type
+      if (
+        schema.properties[propName]['id'] && 
+        schema.properties[propName]['generated'] && 
+        schema.properties[propName]['type'].toLowerCase() === 'string'
+      ) {
+        schema.properties[propName]['useDefaultIdType'] = false;
+      }
       const dbSpecific = schema.properties[propName][dbType] = {
         columnName: item.columnName,
         dataType: item.dataType,
