diff --git a/node_modules/loopback-datasource-juggler/lib/datasource.js b/node_modules/loopback-datasource-juggler/lib/datasource.js
index c163af7..6091e88 100644
--- a/node_modules/loopback-datasource-juggler/lib/datasource.js
+++ b/node_modules/loopback-datasource-juggler/lib/datasource.js
@@ -1744,6 +1744,16 @@ DataSource.prototype.discoverSchemas = function(tableName, options, cb) {
         propertyDetails['index'] = {unique: true};
       }
       schema.properties[propName] = propertyDetails;
+
+      // set useDefaultIdType: false in the case of id property with generated: 1 and has string type
+      if (
+        schema.properties[propName]['id'] && 
+        schema.properties[propName]['generated'] && 
+        schema.properties[propName]['type'].toLowerCase() === 'string'
+      ) {
+        schema.properties[propName]['useDefaultIdType'] = false;
+      }
+      
       const dbSpecific = schema.properties[propName][dbType] = {
         columnName: item.columnName,
         dataType: item.dataType,
