diff --git a/node_modules/loopback-connector-mysql/lib/discovery.js b/node_modules/loopback-connector-mysql/lib/discovery.js
index 615c54d..fe0e21f 100644
--- a/node_modules/loopback-connector-mysql/lib/discovery.js
+++ b/node_modules/loopback-connector-mysql/lib/discovery.js
@@ -166,7 +166,7 @@ function mixinDiscovery(MySQL, mysql) {
         ' numeric_scale AS "dataScale",' +
         ' column_type AS "columnType",' +
         ' is_nullable = \'YES\' AS "nullable",' +
-        ' CASE WHEN extra LIKE \'%auto_increment%\' THEN 1 ELSE 0 END AS "generated"' +
+        ' case when extra like \'%virtual%\' then 1 when extra like \'%stored%\' then 1 when extra LIKE \'%auto_increment%\' THEN 1 else 0 end as "generated"' +
         ' FROM information_schema.columns' +
         ' WHERE table_schema=' + mysql.escape(schema) +
         (table ? ' AND table_name=' + mysql.escape(table) : ''),
@@ -181,7 +181,7 @@ function mixinDiscovery(MySQL, mysql) {
         ' numeric_scale AS "dataScale",' +
         ' column_type AS "columnType",' +
         ' is_nullable = \'YES\' AS "nullable",' +
-        ' CASE WHEN extra LIKE \'%auto_increment%\' THEN 1 ELSE 0 END AS "generated"' +
+        ' case when extra like \'%virtual%\' then 1 when extra like \'%stored%\' then 1 when extra LIKE \'%auto_increment%\' THEN 1 else 0 end as "generated"' +
         ' FROM information_schema.columns' +
         (table ? ' WHERE table_name=' + mysql.escape(table) : ''),
       'table_name, ordinal_position', {});
